pkg/inventory/manager.go

```diff
@@ -14,28 +14,28 @@ import (
 
 // Manager wraps an Inventory with convenience methods that use ObjMetadata.
 type Manager struct {
-	inventory *actuation.Inventory
+	inventory BaseInventory
```

Comment:
Why does this use the struct and not the interface?
If the struct is correct, should we use a pointer to avoid copying every time it's set?

Reply:
The field type could be either the `Inventory` interface or `BaseInventory`, however the instantiation by the `NewManager` constructor would still be `BaseInventory`. This is just used as a simple cache to track object statuses while the applier runs.

It's only set once by the constructor so I don't think it should make a material difference, but let me know what you prefer.

Reply:
I think this using `BaseInventory` instead of `*BaseInventory` here is a subtle bug that may cause unnecessary extra copying of the inventory object. This may have been caught earlier if the field type was the `Inventory` interface, because the BaseInventory methods are all on the pointer.

Also, as a less important point, the Manager using the interface contributes twords proving that the interface is usable as-is and doesn't require using private BaseInventory fields or methods. It could also allow faking, but that's not really necessary in this case because the BaseInventory is so simple.

So I'd prefer using the interface as the struct field type.

Reply:
Looks like if you try to use the interface, you wont be able to access ObjStatuses. So you'll need to call ObjectStatuses/SetObjectStatuses instead. That might require a little refactoring, but that's probably good, to exercise the interface to ensure it's usable.

Reply:
After trying to do this refactor I remembered why I used `BaseInventory` when I originally wrote this. The `Inventory` interface requires implementation of the `Info` interface (Namespace/ID/etc), but this struct is just working with a raw objectRef/objectStatus cache and doesn't know about the metadata of the real inventory object.